support-frontend/assets/helpers/redux/checkout/product/selectors/selectedAmount.ts

```diff
@@ -15,6 +16,18 @@ export function getUserSelectedAmount(state: ContributionsState): number {
 	return priceCardAmountSelected;
 }
 
+export function getAmountCoveringTransactionCost(
+	amount: number,
+	coverTransactionCost?: boolean,
+): number {
+	if (coverTransactionCost) {
```

Comment:
I know we've covered this - but why not put this logic in `getUserSelectedAmount`?

Reply:
Yep, been debating this. One reason was that there were two separate + equivalent get amount from state functions (maybe it would worth consolidating this). The other was that we still need the old amount for keeping choice card selected and displaying the old price. Given the number of places I've had to call this function I think you're right and we should put the logic in `getUserSelectedAmount` and make an exception where we need the old price instead

Reply:
Take a look at - is this better? (Turns out there was a third place that got the contribution amount!) https://github.com/guardian/support-frontend/pull/6282